src/core/string_map.cc

```diff
@@ -101,6 +101,74 @@ sds StringMap::Find(std::string_view key) {
   return GetValue(str);
 }
 
+void StringMap::RandomPair(sds* key, sds* value) {
+  auto itr = begin();
+  for (unsigned int i = 0; i < (rand() % Size()); ++i)
+    ++itr;
+  *key = itr->first;
+  if (value != nullptr)
+    *value = itr->second;
+  return;
+}
```

Comment:
I'm not a fan of C style inside out C++ codebase. Lets return those pointers by value

```c++
pair<sds, sds> RandomPair() {
  auto it = begin();
  std::advane(it, rand % size);
  return it;
}

```

Reply:
very nice, thanks!

Reply:
actually more work needs to be done as StringMap's iterator doesn't support std::advance.

Reply:
@dranikpg i just moved the logic into an += operator, see if that looks ok?

Reply:
You can also add the `difference_type` alias in the Iterator. It's an one liner :smile: